### PR TITLE
test(ARCH-421/cmf): migrate to testing library

### PR DESCRIPTION
**Pull request by jmfrancois**
**What is the problem this PR is trying to solve?**

enzyme is dead: https://dev.to/wojtekmaj/enzyme-is-dead-now-what-ekl

**What is the chosen solution to this problem?**

migrate to testing-library following https://kentcdodds.com/blog/common-mistakes-with-react-testing-library

upgrade ui-scripts to get linters and matchers configured

**Please check if the PR fulfills these requirements**

- [x] The PR have used `yarn changeset` to a request a release from the CI if wanted.
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
- [x] Docs have been added / updated (for bug fixes / features)
- [x] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/cmf/__tests__/App.test.js [hunk 1]
```diff
@@ -15,22 +27,14 @@ describe('CMF App', () => {
 				return {};
 			},
 		};
-		const history = {};
-		const wrapper = shallow(
-			<App store={store} history={history}>
-				<div className="children" />
+		render(
+			<App store={store}>
+				<div className="children">I am a children</div>
```
Comment (frassinier) [inline packages/cmf/__tests__/App.test.js:32]:
```suggestion
				<div className="children">I am a child</div>
```

File: packages/cmf/__tests__/Dispatcher.test.js [hunk 2]
```diff
@@ -91,80 +86,86 @@ describe('Testing <Dispatcher />', () => {
...
+						onClick="noOp"
+						onDoubleClick="existingActionCreator:id"
+					>
+						<a />
+					</Dispatcher>
+				</div>
+			</mock.Provider>,
 		);
-		wrapper.find('a').simulate('click');
+		fireEvent.click(wrapper.container.querySelector('a'));
```
Comment (frassinier) [inline packages/cmf/__tests__/Dispatcher.test.js:102]: https://testing-library.com/docs/react-testing-library/migrate-from-enzyme#test-2-input-texts-must-have-correct-value

> When you use React Testing Library, you should not try to access the DOM like how you would do by document.querySelector API (which you can incidentally use in your tests, but it's not recommended for the reasons stated in this paragraph).

I would encourage you to use text into the anchor instead

File: packages/cmf/__tests__/App.test.js [hunk 3]
```diff
@@ -15,22 +27,14 @@ describe('CMF App', () => {
...
-				<Provider store={store}>
-					<RegistryProvider>
-						<ErrorBoundary fullPage>
-							<div className="children" />
-						</ErrorBoundary>
-					</RegistryProvider>
-				</Provider>,
-			),
-		).toEqual(true);
+		expect(screen.getByText('I am a children')).toBeDefined();
```
Comment (frassinier) [inline packages/cmf/__tests__/App.test.js:35]:
```suggestion
		expect(screen.getByText('I am a child')). toBeTruthy();
```

File: packages/cmf/__tests__/Inject.test.js [hunk 4]
```diff
@@ -1,47 +1,46 @@
...
 		};
-		const Provider = mock.Provider;
 
 		// when
-		const wrapper = mount(
-			<Inject component="MyComponent" extra-props />,
-			Provider.getEnzymeOption(context),
+		const wrapper = render(
+			<mock.Provider registry={registry}>
+				<Inject component="MyComponent" data-test="foo" />
```
Comment (frassinier) [inline packages/cmf/__tests__/Inject.test.js:19]:
```suggestion
				<Inject component="MyComponent" data-testid="foo" />
```

File: packages/cmf/__tests__/Inject.test.js [hunk 5]
```diff
@@ -1,47 +1,46 @@
...
+			</mock.Provider>,
 		);
 
 		// then
-		expect(wrapper.find(MyComponent).equals(<MyComponent extra-props />)).toBe(true);
+		const out = wrapper.container.querySelector('[data-test="foo"]');
```
Comment (frassinier) [inline packages/cmf/__tests__/Inject.test.js:24]:
```suggestion
		const out = screen.getByTestId('foo');
```

File: packages/cmf/__tests__/Inject.test.js [hunk 6]
```diff
@@ -1,47 +1,46 @@
...
-		expect(
-			wrapper
-				.find(Inject.NotFoundComponent)
-				.equals(
-					<Inject.NotFoundComponent error="component not found in the registry: MyComponent" />,
-				),
-		).toBe(true);
+		expect(wrapper.container.querySelector('.alert-danger').textContent).toBe(
+			'component not found in the registry: MyComponent',
+		);
```
Comment (frassinier) [inline packages/cmf/__tests__/Inject.test.js:44]:
```suggestion
		expect(screen.getByText(
			'component not found in the registry: MyComponent',
		)).toBeTruthy();
```

File: packages/cmf/__tests__/cmfModule.merge.test.js [hunk 7]
```diff
@@ -241,14 +238,14 @@ describe('mergeModule', () => {
 
 		// when
 		const { RootComponent } = mergeModules(module1, module2, module3);
-		const wrapper = mount(<RootComponent />);
+		const { container } = render(<RootComponent />);
 
 		// then
-		const mod1 = wrapper.find('#mod1');
+		const mod1 = container.querySelectorAll('#mod1');
```
Comment (frassinier) [inline packages/cmf/__tests__/cmfModule.merge.test.js:244]: The goal of React Testing Library is to test the rendering like a user does, AFAIK.
Can we rely on the content of the #mod1,2,3 instead? 🤔

File: packages/cmf/__tests__/expression.test.js [hunk 8]
```diff
@@ -89,9 +89,13 @@ describe('expression', () => {
...
-		const wrapper = shallow(<WithExpr disabledExpression="test" />, { context });
-		expect(wrapper.props().disabled).toBe(true);
-		expect(wrapper.props().disabled).not.toBe('test');
+		const wrapper = render(
+			<mock.Provider registry={context.registry}>
+				<WithExpr disabledExpression="test" />
+			</mock.Provider>,
+		);
+		const btn = wrapper.container.querySelector('button');
+		expect(btn.getAttribute('disabled')).toBe('');
```
Comment (frassinier) [inline packages/cmf/__tests__/expression.test.js:98]:
```suggestion
		expect(getByText(/Click me/i).closest('button')).toHaveAttribute();
```

File: packages/cmf/src/components/ErrorFeedBack/ErrorFeedBack.component.test.js [hunk 9]
```diff
@@ -24,18 +28,7 @@ describe('Component ErrorFeedBack', () => {
...
-			{
-				message: 'Error message',
-				name: 'Error name',
-				stack: 'Error stack',
-			},
-		];
-		const wrapper = shallow(<Component errors={errors} />);
-		expect(wrapper.find('Fragment').length).toBe(1);
+		const wrapper = render(<Component errors={errors} fullPage />);
+		expect(wrapper.container.querySelector('.col-md-6').style.marginTop).toBe('200px');
```
Comment (frassinier) [inline packages/cmf/src/components/ErrorFeedBack/ErrorFeedBack.component.test.js:32]: I didn't get the added value of that test

Comment (jmfrancois) [reply packages/cmf/src/components/ErrorFeedBack/ErrorFeedBack.component.test.js:32]: lets remove it !

File: packages/cmf/package.json [hunk 10]
```diff
@@ -63,13 +63,12 @@
...
     "@talend/scripts-core": "^11.3.0",
     "@talend/scripts-preset-react-lib": "^9.9.3",
-    "enzyme": "^3.11.0",
+    "@testing-library/react": "^12.1.2",
     "jest-in-case": "^1.0.2",
     "jsdoc": "^3.6.7",
     "node-fetch": "^2.6.6",
     "react": "^16.14.0",
     "react-dom": "^16.14.0",
-    "react-test-renderer": "^16.14.0",
```
Comment (jmfrancois) [inline packages/cmf/package.json:72]: this was not used at all in CMF 🤔

File: packages/cmf/src/Inject.component.js [hunk 11]
```diff
@@ -35,7 +35,7 @@ function Inject({ component, ...props }) {
 		const Component = componentAPI.get(component, context);
 		return <Component {...props} />;
 	} catch (error) {
-		return <NotFoundComponent error={error.message} />;
+		return <NotFoundComponent error={error} />;
```
Comment (jmfrancois) [inline packages/cmf/src/Inject.component.js:38]: this is the fix discovered using tests and why there is a changeset in the PR

File: .changeset/pretty-cycles-exercise.md [hunk 12]
```diff
@@ -0,0 +1,5 @@
+---
+'@talend/react-cmf': patch
+---
+
+Inject display error in case the component is not found
```
Comment (frassinier) [inline .changeset/pretty-cycles-exercise.md:5]: Is it the real purpose of that change?

**Review by frassinier — APPROVED** [review]
Great move!

**Review by Benrajalu — APPROVED** [review]

**Pull request merged**